starlette/requests.py

```diff
@@ -265,8 +267,11 @@ async def form(self) -> FormData:
                 self._form = FormData()
         return self._form
 
+    def form(self) -> AwaitableOrContextManager[FormData]:
```

Comment:
Should this be async?

```suggestion
    async def form(self) -> AwaitableOrContextManager[FormData]:
```

I understand it would work either way just because the awaitable thing is returned, but maybe it could help to make it async, to make it explicit in the function that it returns something to be awaited. What do you think?